### ISSUE TITLE
Prettier range ignore doesn't work for codeblock

### ISSUE THREAD

**Issue by bennycode** [prettier/prettier#14374]
**Prettier 2.6.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAhAWkwAgA4BOcMMAlnAZqQOZQRGYDOMAhgTDtgHwA6UwAUhyQAJnABGAGwhgA1jkkso1JAE8WAW0nCIY6ggB0qrTgEBfPnkkBXaqSiMkfHJxwsRIp1BcvIsBDBIOADkpBp49BwwqnhwOMA4ACIy1hoBAHK6cWY4AGYEEBo4PCDUBCx4ABYAjpIA9IrK1iz6dSzMJQDcwc6u0bGMYASkeDC9uP1wg8OjmBCx5WTQjOM4k9MjMJhELGBbFRCS0nyCOAgiohLScqYWUBjY+EQk5JQ0dAznnJhcIAA0IHmSwcyFAbAKAHcAApsBCMZAgFiSCEsVTwgHicpyYgAZU0cAAMvY4MhckjGHAMVjZLi8Lt7NRkDACNZKSA4BpxHAPNyCUpqM19AAxegaFgvZQIljWGAQf4gSowLQAdUqpHgjDpYDgOLh6tIADd1aoEWBGOiQPYKewoeVqGLSeS2QArRgADxxDMkcAAitYIPBHZIKQC6QRrQixQRZCIIBCoPLCPYYMrSCIYJVkAAOAAMoYKFOV5TwCMIUwoBpJAOq-vgtvm8JQ7UwUDg3O58qINdIRFtLQdSDJwbZFI0pCZLJHXt9tZJg6dANY4lT6czSAATIvyqRJAyAMKFAfsxgAVnl1gpABUWOJG0OQyADayAJJQMSwHFDTYAQTfOOi3pBhSZhmEAA)

```sh
--parser markdown
```

**Input:**

```markdown
<!-- prettier-ignore-start -->
{% codeblock lang:yaml codegen.yml %}
plugins:
  - add:
    content: 'import type { DocumentNode } from "graphql/language/ast";'
  - typescript
  - typescript-operations
  - typescript-react-apollo
{% endcodeblock %}
<!-- prettier-ignore-end -->
```

**Output:**

```markdown
<!-- prettier-ignore-start -->

{% codeblock lang:yaml codegen.yml %}
plugins:

- add:
  content: 'import type { DocumentNode } from "graphql/language/ast";'
- typescript
- typescript-operations
- typescript-react-apollo
{% endcodeblock %}
<!-- prettier-ignore-end -->

```

**Expected behavior:**

I expect the [range ignore](https://prettier.io/docs/en/ignore.html#range-ignore) functionality to not format code within the `{% codeblock %}` statement.

**Comment by BernhardBln** [comment]
@kachkaev fair enough, but could it be mentioned in the docs (strongly highlighted!)? I spend a lot of time until I found this comment.

https://prettier.io/docs/en/ignore.html#range-ignore

The docs even suggest it's ok to put it without newline in the example (probably works in that case, but apparently not in general...)

<img width="811" alt="image" src="https://github.com/prettier/prettier/assets/4759839/99fa3734-01e2-4409-8464-2e7ecb0995d5">
